odoo/tools/pycompat.py

```diff
@@ -3,37 +3,40 @@
 import csv
 import codecs
 import io
+import typing
+import warnings
 
 _reader = codecs.getreader('utf-8')
 _writer = codecs.getwriter('utf-8')
 
 
 def csv_reader(stream, **params):
+    warnings.warn("Deprecated since Odoo 18.0: can just use `csv.reader` with a text stream or use `TextIOWriter` or `codec.getreader` to transcode.", DeprecationWarning, 2)
     assert not isinstance(stream, io.TextIOBase),\
         "For cross-compatibility purposes, csv_reader takes a bytes stream"
     return csv.reader(_reader(stream), **params)
 
 
 def csv_writer(stream, **params):
+    warnings.warn("Deprecated since Odoo 18.0: can just use `csv.writer` with a text stream or use `TextIOWriter` or `codec.getwriter` to transcode.", DeprecationWarning, 2)
     assert not isinstance(stream, io.TextIOBase), \
         "For cross-compatibility purposes, csv_writer takes a bytes stream"
     return csv.writer(_writer(stream), **params)
 
 
-def to_text(source):
-    """ Generates a text value (an instance of text_type) from an arbitrary
-    source.
+def to_text(source: typing.Any) -> str:
```

Comment:
This function can be moved to misc where near `str2bool`, these are just small helpers and this way that package may be removed in v18.